### PR TITLE
Only allow 1 source. Add tagPrefix field to source.

### LINKED ISSUE #577: Add support for --attachment-tag-prefix attestations.
**Description**

If you create attestations with the tag prefix, then we can't find them. We should add support for those in the API.

### PR DESCRIPTION
**Pull request by vaikas**
#### Summary
Fixes #577 
Fixes: #360 

As described in #360, last entry in the list wins, which is surprising and wrong. So, limit Source to only have 1 entry in the list for correctness.

Add source.tagPrefix field that allows one to get verify parity when signing and attesting with: `--attachment-tag-prefix` 

#### Release Note
Support tag-prefix for verifying signature / attestation.
**Breaking Change** You can only have 1 source specified for Authority.

### CODE REVIEW

File: test/testdata/policy-controller/e2e/cip-keyless-with-source.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ spec:
       - issuerRegExp: .*kubernetes.default.*
         subjectRegExp: .*kubernetes.io/namespaces/default/serviceaccounts/default
     source:
-    - oci: registry.local:5000/policy-controller/demo
+    # DO NOT CHECK THIS CHANGE IN.
```
Comment (vaikas) [inline test/testdata/policy-controller/e2e/cip-keyless-with-source.yaml:29]: Great job @vaikas  on this one :) One job :)

**Comment by vaikas** [comment]
There are lots of unrelated e2e errors.

**Review by hectorj2f — COMMENTED** [review]
lgtm so far!

**Comment by hectorj2f** [comment]
@vaikas Please, rebase! We have some fixes for these failing tests.

**Comment by vaikas** [comment]
Done! Though, I think it picked up the test fixes judging by the greennness on the last run :)

**Comment by hectorj2f** [comment]
@vaikas I think we managed to solve the issue with `cgr.dev` :D

**Review by hectorj2f — APPROVED** [review]

**Pull request merged**